p2p/src/simulated/ingress.rs

```diff
@@ -93,21 +97,37 @@ impl<P: PublicKey> Oracle<P> {
     ///
     /// By default, the peer will not be linked to any other peers. If a peer is already
     /// registered on a given channel, it will return an error.
-    ///
-    /// Bandwidth can be specified for the peer's egress (upload) and ingress (download)
-    /// rates in bytes per second. `None` means unlimited bandwidth.
     pub async fn register(
         &mut self,
         public_key: P,
         channel: Channel,
-        egress_bps: Option<usize>,
-        ingress_bps: Option<usize>,
     ) -> Result<(Sender<P>, Receiver<P>), Error> {
         let (sender, receiver) = oneshot::channel();
         self.sender
             .send(Message::Register {
                 public_key,
                 channel,
+                result: sender,
+            })
+            .await
+            .map_err(|_| Error::NetworkClosed)?;
+        receiver.await.map_err(|_| Error::NetworkClosed)?
+    }
+
+    /// Set bandwidth limits for a peer.
+    ///
+    /// Bandwidth can be specified for the peer's egress (upload) and ingress (download)
+    /// rates in bytes per second. `None` means unlimited bandwidth.
```

Comment:
IMO we should no longer support `None`. If you are calling `set_bandwidth`, you should provide specific values.

Reply:
I understand this prevents specifying ingress and not egress (or vice-versa) but suspect anyone specifying this will set both.